client/gutenberg/editor/controller.js

```diff
@@ -331,6 +331,10 @@ export function redirectToPermalinkIfLoggedOut( context, next ) {
 	) {
 		return next();
 	}
+
+	if ( ! context.params.post ) {
```

Comment:
There are a few things to iron out in this logic IMHO - solely checking if the `post` param is falsy doesn't seem enough to me. 

Looking at the `getPostID()` function that's used while handling the `post` controller logic, it appears that we have some possible string values for the `post` param. Particularly, it can be `new` (when creating a new post) and can also be `home` (if editing the home page). There may be additional possible values - that's worth double-checking. That being said, we likely need to handle those values specifically, as done in `getPostID()`, particularly because I don't see them handled in the `editor/redirect` endpoint.

Furthermore, given that we're handling numbers only and we've excluded the abovementioned scenario of a string in the `post` param, we might want to do a `parseInt( context.params.post, 10 )` and only redirect if the result is actually an integer.

Reply:
It's easy for me to imagine that someone could come around and confuse `.post` and `.path` and assume that _this_ fix was a typo.

Inverting the logic to only redirect the location on a viable post/page/single view could help with that, as could giving a name to the check, vs. relying on implicit logic without explanation.

```js
const postId = parseInt( context.params.post, 10 );
const linksToSingleView = postId > 0;

if ( linksToSingleView ) {
	// Redirect the logged??? user to the permalink of the post, page, custom …
	// …
	window.location = `…`;
}

return next();
```